### ISSUE TITLE
Improve Shader class API by adding the same methods of the Visual Shader API

### ISSUE THREAD

**Issue by paddy-exe** [godotengine/godot-proposals#5345]
### Describe the project you are working on

I had an idea to create a User Interface addon based on shaders

### Describe the problem or limitation you are having in your project

Currently it is impossible to expose a drop-down option via the shading language. However this is possible in C++ in the engine code.
This limits the possibility to procedurally change the shader properties through GDScript.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I understand the draw-back of exposing this to the shader language due to longer compilation time for the shaders. Doing this however through GDScript just like it's done via C++ to change the shader code based on the options in Visual Shaders or a BaseMaterial should negate or at least lessen the compilation time.

This would enable to create inspectors similar to this (I only mean the drop-down option):
![image](https://user-images.githubusercontent.com/38077837/188510000-a1168f05-bc39-4d7b-8209-b4fbfaf3f3e3.png)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I would propose to add similar / the same API of the VisualShader class to the Shader class.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can't. The proposal is about exposing this to GDScript in the first place

### Is there a reason why this should be core and not an add-on in the asset library?

See above

**Comment by paddy-exe** [comment]
For reference here are the links to the VisualShader API and to the Shader API:
[Shader API](https://docs.godotengine.org/en/stable/classes/class_shader.html)
[Visual Shader API](https://docs.godotengine.org/en/stable/classes/class_visualshader.html)

**Comment by YuriSizov** [comment]
@paddy-exe Did you close this on purpose?

**Comment by paddy-exe** [comment]
> @paddy-exe Did you close this on purpose?

Yes, sorry for closing this without an explanation. QbieShay made me aware that I could do this already with a bit more work on my side

**Comment by Zireael07** [comment]
@paddy-exe Would be worth sharing how you do it, e.g. on a blog or Godot subreddit

**Comment by paddy-exe** [comment]
@Zireael07 I will! First I would need to create the add-on though. And should be like QbieShay's example code:
```
shader_code = global_string + "void vertex(){" + vertex_string + "} void fragment(){" + fragment_string + "} void light(){" + light_code + "}"
```
